### ISSUE TITLE
connectToStores may cause component to render multiple times

### ISSUE THREAD

**Issue by jlecomte** [yahoo/fluxible-addons-react#20]
When using `connectToStores` to connect a React component to multiple stores which can respond to the same action, the React component will be rendered multiple times when that action is dispatched. Beyond the obvious performance penalty, this can also add complexity to the React component's implementation. All the gory details and a simplistic patch are available at:

https://github.com/jlecomte/fluxible-connectToStores-issue-testcase

**Comment by mridgway** [comment]
React will not batch updates if they happen outside of the original event loop fired from one of its lifecycle events or click handlers. Looks like they are trying to find a way to make batching the default, but has not been solved yet. In the mean time, we would have to find a better way to batch updates that are happening within a dispatch loop.

Using asynchrony in this manner can actually cause other bugs because the value of `this.state` will not be updated, so if there are other places that depend on `this.state` we may cause inadvertent and hard to  find bugs.

React does provide `require(React/addons`).batchedUpdates` (moved in React 0.14), which allows you to batch synchronous `setState` calls together. We could provide a plugin that wraps `dispatch` with a call to `batchedUpdates` which would ensure that render is only called once per render. Here's what I was playing around with to test:

```js
var context = app.createContext();
var actionContext = context.getActionContext();
var oldDispatch = actionContext.dispatch;
actionContext.dispatch = function () {
    var args = arguments;
    React.addons.batchedUpdates(function () {
        oldDispatch.apply(actionContext, args);
    });
};
```

**Comment by mridgway** [comment]
The plugin has been released in version 0.1.7. Let me know if this resolves your issue.

**Comment by geekyme** [comment]
@mridgway do i have to use react 0.14.0 to use that functionality?
